### PR TITLE
feat(fgs/function): function support parameter 'tags'

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Function support parameter 'tags'.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. function support parameter 'tags'
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/fgs' TESTARGS='-run=TestAccFgsV2Function_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/fgs -v -run=TestAccFgsV2Function_basic -timeout 360m -parallel 4
=== RUN   TestAccFgsV2Function_basic
=== PAUSE TestAccFgsV2Function_basic
=== CONT  TestAccFgsV2Function_basic
--- PASS: TestAccFgsV2Function_basic (65.17s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/fgs       65.266s
```

### CODE REVIEW

File: docs/resources/fgs_function.md [hunk 1]
```diff
@@ -207,3 +209,21 @@ Functions can be imported using the `id`, e.g.
...
 ```
+
+Note that the imported state may not be identical to your resource definition, due to the attribute missing from the
+API response. The missing attributes are:
+`app`, `func_code`, `agency`, `tags"`.
+It is generally recommended running `terraform plan` after importing a function.
+You can then decide if changes should be applied to the function, or the resource definition should be updated to align
+with the function. Also you can ignore changes as below.
+
+```
```
Comment (deer-hang) [inline docs/resources/fgs_function.md]: add hcl

File: huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go [hunk 2]
```diff
@@ -351,6 +354,15 @@ func resourceFgsFunctionV2Create(d *schema.ResourceData, meta interface{}) error
 		}
 	}
 
+	if tagList, ok := d.GetOk("tags"); ok {
+		opts := function.TagsActionOpts{
+			Tags: utils.ExpandResourceTags(tagList.(map[string]interface{})),
+		}
+		if err := function.CreateResourceTags(fgsClient, d.Id(), opts); err != nil {
+			log.Printf("[ERROR] failed to add tags to FunctionGraph function (%s): %s", d.Id(), err)
```
Comment (deer-hang) [inline huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go]: The create error should return `diag.Errorf("....")`

Comment (Lance52259) [reply huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go]: some region does not support this function.

File: huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go [hunk 3]
```diff
@@ -463,6 +475,34 @@ func resourceFgsFunctionV2Read(d *schema.ResourceData, meta interface{}) error {
...
+		nMap        = nRaw.(map[string]interface{})
+		functionUrn = d.Id()
+	)
+
+	if len(oMap) > 0 {
+		opts := function.TagsActionOpts{
+			Tags: utils.ExpandResourceTags(oMap),
+		}
+		if err := function.CreateResourceTags(client, functionUrn, opts); err != nil {
+			return fmt.Errorf("failed to add tags to FunctionGraph function (%s): %s", functionUrn, err)
```
Comment (deer-hang) [inline huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go]: oMap means remove old tags?
Maybe the error message and the function is not correct

File: huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go [hunk 4]
```diff
@@ -463,6 +475,34 @@ func resourceFgsFunctionV2Read(d *schema.ResourceData, meta interface{}) error {
...
+		}
+	}
+
+	if len(nMap) > 0 {
+		opts := function.TagsActionOpts{
+			Tags: utils.ExpandResourceTags(nMap),
+		}
+		if err := function.CreateResourceTags(client, functionUrn, opts); err != nil {
+			return fmt.Errorf("failed to delete tags from FunctionGraph function (%s): %s", functionUrn, err)
```
Comment (deer-hang) [inline huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go]: ditto

File: huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go (hunk 2 above)
Comment (deer-hang) [reply huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go]: How about add some description for this:
// Some region not support Funtion Tags, ignore errors.

File: docs/resources/fgs_function.md (hunk 1 above)
Comment (Lance52259) [reply docs/resources/fgs_function.md]: done.

File: huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go]: emmm, throw the error is right, if corresponding region is not support tags parameter, this parameter should not be used int the script.

File: huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go (hunk 3 above)
Comment (Lance52259) [reply huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go]: fixed.

File: huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go (hunk 4 above)
Comment (Lance52259) [reply huaweicloud/services/fgs/resource_huaweicloud_fgs_function.go]: fixed.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/fgs' TESTARGS='-run=TestAccFgsV2Function_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/fgs -v -run=TestAccFgsV2Function_basic -timeout 360m -parallel 4
=== RUN   TestAccFgsV2Function_basic
=== PAUSE TestAccFgsV2Function_basic
=== CONT  TestAccFgsV2Function_basic
--- PASS: TestAccFgsV2Function_basic (58.63s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/fgs       58.780s
```

**Pull request merged**